src/main/java/com/spider/solomon/reporting/ReportController.java

```diff
@@ -0,0 +1,26 @@
+package com.spider.solomon.reporting;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import jakarta.annotation.Resource;
+
+/**
+ * Endpoints for general reporting
+ */
+@RestController
+@RequestMapping("/reports")
+public class ReportController {
+	@Resource
+	TrendAnalyzer trendAnalyzer;
+
+	/**
+	 * Returns the trend of the total sales data
+	 */
+	@RequestMapping(value = "/trend", method = RequestMethod.GET)
+	public String trend () {
```

Comment:
Could this be considered a premature optimization? Why or why not?

Reply:
If it is almost certain the endpoint has a single purpose, yes it can be considered premature.  I definitely would pose the question to whoever writes requirements.

If somewhere on the roadmap there is some possibility of introducing things like `newCustomersTrend` it would be useful to make it an object to reduce the re-work involved of a versioned endpoint or having to coordinate simultaneous UI and backend server deployments to match any schema changes.